sway-fmt-v2/src/items/item_impl/tests.rs

```diff
@@ -0,0 +1,153 @@
+    ($scope:ident $desired_output:expr, $($name:ident $y:expr),+) => {
+        $(
+        paste! {
+            #[test]
+            fn [<$scope _ $name>] () {
+                let formatted_code = format_code($y);
+                let changeset = diff_lines(&formatted_code, $desired_output);
+                let diff = changeset.diff();
+                let count_of_updates = diff.len();
+                if count_of_updates != 0 {
+                    println!("FAILED: {count_of_updates} diff items.");
+                }
+                for diff in diff {
+                    match diff {
+                        DiffOp::Equal(old) => {
+                            for o in old {
+                                println!("{}", o)
+                            }
+                        }
+                        DiffOp::Insert(new) => {
+                            for n in new {
+                                println_green(&format!("+{}", n));
+                            }
+                        }
+                        DiffOp::Remove(old) => {
+                            for o in old {
+                                println_red(&format!("-{}", o));
+                            }
+                        }
+                        DiffOp::Replace(old, new) => {
+                            for o in old {
+                                println_red(&format!("-{}", o));
+                            }
+                            for n in new {
+                                println_green(&format!("+{}", n));
+                            }
+                        }
+                    }
+                }
+                assert_eq!(&formatted_code, $desired_output)
+            }
+        }
+    )+
+}
+}
+
+fmt_test!(  impl_with_nested_items
+"impl AuthTesting for Contract {
+    fn returns_msg_sender(expected_id: ContractId) -> bool {
+        let result: Result<Identity, AuthError> = msg_sender();
+        let mut ret = false;
+        if result.is_err() {
+            ret = false;
+        }
+        let unwrapped = result.unwrap();
+        match unwrapped {
+            Identity::ContractId(v) => {
+                ret = true
+            }
+            _ => {
+                ret = false
+            }
+        }
+        ret
+    }
+}",
+            intermediate_whitespace
+"impl AuthTesting for Contract {
+    fn returns_msg_sender(expected_id: ContractId) -> bool {
+        let result: Result<Identity, AuthError> = msg_sender();
+        let mut ret = false;
+        if result.is_err() {
+            ret = false;
+        }
+        let unwrapped = result.unwrap();
+        match unwrapped {
+            Identity::ContractId(v) => {ret = true}
+            _ => {ret = false}
+        }
+        ret
+    }
+}"
+);
+
+fmt_test!(  normal_with_generics
+"impl<T> Option<T> {
+    fn some(value: T) -> Self {
+        Option::Some::<T>(value)
+    }
+    fn none() -> Self {
+        Option::None::<T>()
+    }
```

Comment:
I think each function should be separated by a new line

Reply:
In this instance since it's purpose is to test the generics themselves, the formatting of the inner functions is irrelevant. I could remove them if that makes it more clear though!